packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/locked-point-settings.tsx

```diff
@@ -104,28 +105,39 @@ const LockedPointSettings = (props: Props) => {
 
     const isDefiningPoint = !onMove && !onRemove;
 
-    /**
-     * Get a prepopulated aria label for the point.
-     *
-     * If the point has no labels, the aria label will just be
-     * "Point at (x, y)".
-     *
-     * If the point has labels, the aria label will be
-     * "Point at (x, y) with label1, label2, label3".
-     */
-    function getPrepopulatedAriaLabel() {
-        let visiblelabel = "";
-        if (labels && labels.length > 0) {
-            visiblelabel += ` ${labels.map((l) => l.text).join(", ")}`;
-        }
-        let str = `Point${visiblelabel} at (${coord[0]}, ${coord[1]})`;
+    const [prepopulatedAriaLabel, setPrepopulatedAriaLabel] =
+        React.useState("");
+
+    React.useEffect(() => {
+        /**
+         * Get a prepopulated aria label for the point, with the math
+         * details converted into spoken words.
+         *
+         * If the point has no labels, the aria label will just be
+         * "Point at (x, y)".
+         *
+         * If the point has labels, the aria label will be
+         * "Point at (x, y) with label1, label2, label3".
+         */
+        async function getPrepopulatedAriaLabel() {
+            let visiblelabel = "";
+            if (labels && labels.length > 0) {
+                visiblelabel += ` ${labels.map((l) => l.text).join(", ")}`;
+            }
 
-        const pointAppearance =
-            generateLockedFigureAppearanceDescription(pointColor);
-        str += pointAppearance;
+            let str = await generateSpokenMathDetails(
+                `Point${visiblelabel} at (${coord[0]}, ${coord[1]})`,
+            );
 
-        return str;
-    }
+            const pointAppearance =
+                generateLockedFigureAppearanceDescription(pointColor);
+            str += pointAppearance;
+
+            return str;
+        }
+
+        getPrepopulatedAriaLabel().then(setPrepopulatedAriaLabel);
```

Comment:
There's a race condition here because the last `getPrepopulatedAriaLabel` promise to resolve might not be one from the most recent render. This could result in an aria label being generated with the wrong coords, labels, or pointColor.

I think we could guard against that race condition by doing this in the useEffect callback:

```ts
let canceled = false;
getPrepopulatedAriaLabel().then((label) => {
  !canceled && setPrepopulatedAriaLabel(label);
})

return () => canceled = true;
```

The annoying thing is that this is really hard to test.

What we really want here is something like a `useMemo`, but unfortunately `useMemo` doesn't work when the function we want to memoize is async. I don't think it would be too hard to write a `useAsyncMemo` hook ourselves, though.

Then we could just write this:

```ts
const prepopulatedAriaLabel = useAsyncMemo(
  /* initial value */ "",
  () => getPrepopulatedAriaLabel(coord, labels, pointColor),
  [coord, labels, pointColor],
);
```